### PR TITLE
Fix `github` formatter for missing final newline

### LINKED ISSUE #6821: When using GitHub Formatter with pnpm, error codes are unintentionally combined in the last violation
### What steps are needed to reproduce the bug?

add package.json script `"lint:style": "stylelint \"**/*.css\""`

```json
{
  "scripts": {
    "lint:style": "stylelint \"**/*.css\""
  }
}
```

then run it

```shell
pnpm lint:style --formatter github
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": ["stylelint-config-standard"]
}
```

### How did you run Stylelint?

via package.json script

```
stylelint "**/*.css" "--formatter" "github" "**/*.css"
```

### Which version of Stylelint or dependencies are you using?

15.6.0

### What did you expect to happen?

```shell
> my-app@0.0.0 lint:style /Users/user/my-app
> stylelint "**/*.css" "--formatter" "github"

::error file=/Users/user/my-app/packages/ui/styles/index.css,line=12,col=1,endLine=12,endColumn=21,title=Stylelint problem::Unexpected empty line before at-rule (at-rule-empty-line-before) [maybe fixable] - https://stylelint.io/user-guide/rules/at-rule-empty-line-before
::error file=/Users/user/my-app/packages/ui/styles/index.css,line=12,col=9,endLine=12,endColumn=20,title=Stylelint problem::Expected ""reset.css"" to be "url("reset.css")" (import-notation) [maybe fixable] - https://stylelint.io/user-guide/rules/import-notation
ELIFECYCLE  Command failed with exit code 2.
```

### What actually happened?

```shell
> my-app@0.0.0 lint:style /Users/user/my-app
> stylelint "**/*.css" "--formatter" "github"

::error file=/Users/user/my-app/packages/ui/styles/index.css,line=12,col=1,endLine=12,endColumn=21,title=Stylelint problem::Unexpected empty line before at-rule (at-rule-empty-line-before) [maybe fixable] - https://stylelint.io/user-guide/rules/at-rule-empty-line-before
::error file=/Users/user/my-app/packages/ui/styles/index.css,line=12,col=9,endLine=12,endColumn=20,title=Stylelint problem::Expected ""reset.css"" to be "url("reset.css")" (import-notation) [maybe fixable] - https://stylelint.io/user-guide/rules/import-notation ELIFECYCLE  Command failed with exit code 2.
```

### Does the bug relate to non-standard syntax?

_No response_

### Proposal to fix the bug

- add newline at the end of the output

### PR DESCRIPTION
**Pull request by konomae**
- close #6821 

To prevent unexpected string concatenation

### CODE REVIEW

File: lib/formatters/__tests__/githubFormatter.test.js [hunk 1]
```diff
@@ -2,58 +2,75 @@
...
-		},
-	];
-	const returnValue = {
-		ruleMetadata: {
-			foo: { url: 'https://stylelint.io/rules/foo' },
-			bar: { url: 'https://stylelint.io/rules/bar', fixable: true, deprecated: true },
-		},
-	};
+describe('githubFormatter', () => {
+	test('outputs no warnings', () => {
```
Comment (konomae) [inline lib/formatters/__tests__/githubFormatter.test.js:6]: Added case with no violations

File: lib/formatters/__tests__/githubFormatter.test.js [hunk 2]
```diff
@@ -2,58 +2,75 @@
...
+			},
+		};
+
+		expect(githubFormatter(results, returnValue))
+			.toBe(`::error file=path/to/file.css,line=1,col=2,endLine=1,endColumn=5,title=Stylelint problem::Unexpected "foo" (foo) - https://stylelint.io/rules/foo
 ::warning file=a.css,line=10,col=20,title=Stylelint problem::Unexpected "bar" (bar) [maybe fixable, deprecated] - https://stylelint.io/rules/bar
 ::error file=a.css,line=20,col=1,title=Stylelint problem::Cannot parse foo (foo-error)
-::error file=a.css,line=20,col=3,title=Stylelint problem::Anonymous error`);
+::error file=a.css,line=20,col=3,title=Stylelint problem::Anonymous error
+`);
```
Comment (konomae) [inline lib/formatters/__tests__/githubFormatter.test.js:74]: The indentation change makes the modification appear larger, but the only thing that changes the result is the line break here.

File: lib/formatters/githubFormatter.js [hunk 3]
```diff
@@ -11,19 +11,21 @@ module.exports = function githubFormatter(results, returnValue) {
...
+		return warnings.map(({ line, column, endLine, endColumn, text, severity, rule }) => {
+			const msg = buildMessage(text, metadata[rule]);
+
+			return endLine === undefined
+				? `::${severity} file=${source},line=${line},col=${column},title=${title}::${msg}`
+				: `::${severity} file=${source},line=${line},col=${column},endLine=${endLine},endColumn=${endColumn},title=${title}::${msg}`;
+		});
+	});
+
+	lines.push('');
```
Comment (konomae) [inline lib/formatters/githubFormatter.js:26]: This is the only place where the logic has changed.

**Review by ybiquitous — APPROVED** [review]
Thank you! LGTM 👍🏼

**Comment by konomae** [comment]
Thanks for the review and the changelog entry!

**Pull request merged**